### DISCUSSION TITLE
Understanding Autarky and Ratio

### DISCUSSION THREAD

**Discussion by Idskov** [slipx06/sunsynk-power-flow-card#340] [Q&A] [answered]
I can't seem to figure out how Autarky and Ratio is calculated, how to understand it and what the difference is between Autarky power and energy.
Could anyone give me the [ELI5](https://www.bitdegree.org/crypto/learn/crypto-terms/what-does-eli5-mean#:~:text=ELI5%20is%20an%20acronym%20for,about%20complex%20terminology%20and%20concepts.)-version?

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/64523817/9f99c0c9-1381-42f5-8d34-4feb71ea340f)

**Comment by slipx06** [comment, marked as answer]
Hi. This is from the docs

Display autarky and ratio as a percentage using either realtime power or daily energy values. Set to no to hide (energy/power/no).
Autarky is the percentage of self sufficiency through Home Production. Ratio is the percentage of produced electricity used by the home.
It is calculated based on the formula below and borrowed from the [Power Distribution Card](https://github.com/JonahKr/power-distribution-card)

Autarky in Percent = Home Production / Home Consumption
Ratio in Percent = Home Consumption / Home Production

If using energy values
```
productionEnergy = stateDayPVEnergy.toNum() + stateDayBatteryDischarge.toNum();
consumptionEnergy = stateDayLoadEnergy.toNum() + stateDayBatteryCharge.toNum();
```

If using power values
```
let productionPower =
            totalPV +
            Utils.toNum(`${batteryPower > 0 ? batteryPower : 0}`) +
            Utils.toNum(`${auxPower < 0 ? auxPower * -1 : 0}`);
let consumptionPower =
            essentialPower +
            nonessentialPower +
            Utils.toNum(`${auxPower > 0 ? auxPower : 0}`) +
            Utils.toNum(`${batteryPower < 0 ? batteryPower * -1 : 0}`);
```

**Reply by Idskov** [reply]
Thank you @slipx06 
Don't know how I failed to find that...
